.github/workflows/notebook-test-suite.yml

```diff
@@ -0,0 +1,143 @@
+# is provided which is triggered when a Harmonny server deployment occurs.
+#
+# Required settings in the repository:
+# Environments (exactly the following strings):
+# - prod
+# - uat
+#
+# Secrets in each environment:
+# - EDL_USER
+# - EDL_PASSWORD
+#
+# Note: A workflow_call event can only have inputs of type boolean, number or
+# string.
+name: Run Jupyter notebook based test suite
+
+on:
+  workflow_call:
+    inputs:
+      harmony-environment:
+        required: true
+        type: string
+      docker-image-tag:
+        required: false
+        type: string
+        default: 'latest'
+
+  repository_dispatch:
+    types: [all,geoloco,harmony-gdal-adapter,harmony-netcdf-to-zarr,harmony-service-example,hoss,hybig,net2cog,swath-projector,trajectory-subsetter]
+    inputs:
+      harmony-environment:
+        required: true
+        type: choice
+        options:
+          - uat
+          - prod
+      docker-image-tag:
+        required: false
+        type: string
+        default: 'latest'
+
+jobs:
+  base:
+    environment: ${{ github.event.client_payload.harmony-environment }}
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.create-matrix.outputs.matrix }}
+      harmony_host_url: ${{ env.HARMONY_HOST_URL }}
+      run_url: ${{ env.RUN_URL }}
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+
+      - name: Get the URL of the current workflow run
+        run: echo "RUN_URL=${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}" >> $GITHUB_ENV
+
+      - name: Set HARMONY_HOST_URL
+        run: |
+          if [ "${{ github.event.client_payload.harmony-environment }}" == "uat" ]; then
+            echo "HARMONY_HOST_URL=https://harmony.uat.earthdata.nasa.gov"
+            echo "HARMONY_HOST_URL=https://harmony.uat.earthdata.nasa.gov" >> $GITHUB_ENV
+          elif [ "${{ github.event.client_payload.harmony-environment }}" == "prod" ]; then
+            echo "HARMONY_HOST_URL=https://harmony.earthdata.nasa.gov"
+            echo "HARMONY_HOST_URL=https://harmony.earthdata.nasa.gov" >> $GITHUB_ENV
+          else
+            echo "HARMONY_HOST_URL=https://harmony.sit.earthdata.nasa.gov"
+            echo "HARMONY_HOST_URL=https://harmony.sit.earthdata.nasa.gov" >> $GITHUB_ENV
+          fi
+
+      - name: Create matrix configuration from Environment Variable
+        id: create-matrix
+        run: |
+          # Use the event type to select the appropriate service list
+          SERVICES_TESTS=$(cat ./config/services_tests_config_${{ github.event.client_payload.harmony-environment }}.json | jq -r '."${{ github.event.action }}"')
+          # Split the selected service list into an array format and set it as the matrix
+          MATRIX=$(echo ",$SERVICES_TESTS" | tr ',' '\n' | awk '{$1=$1;print}' | jq -c -R '[inputs] | {service: .}')
+          echo "matrix=$MATRIX" >> $GITHUB_OUTPUT
```

Comment:
bash wizardry...

Reply:
I created a bash script and test it out. I looks like the following:
```
#!/bin/bash
SELECTED_SERVICES="hoss   , variable-subsetter"
tests=$(echo ",$SELECTED_SERVICES" | tr ',' '\n' | awk '{$1=$1;print}' | jq -c -R '[inputs] | {service: .}')
echo "$tests"
```

Reply:
It outputs: `{"service":["hoss","variable-subsetter"]}`

Reply:
quite obvious when I'm told. 🤦